modules/eslint-plugin/spec/rules/component-store/avoid-mapping-component-store-selectors.spec.ts

```diff
@@ -0,0 +1,141 @@
+import type {
+  ESLintUtils,
+  TSESLint,
+} from '@typescript-eslint/experimental-utils';
+import { fromFixture } from 'eslint-etc';
+import * as path from 'path';
+import rule, {
+  messageId,
+} from '../../../src/rules/component-store/avoid-mapping-component-store-selectors';
+import { ruleTester } from '../../utils';
+
+type MessageIds = ESLintUtils.InferMessageIdsTypeFromRule<typeof rule>;
+type Options = ESLintUtils.InferOptionsTypeFromRule<typeof rule>;
+type RunTests = TSESLint.RunTests<MessageIds, Options>;
+
+const validConstructor: () => RunTests['valid'] = () => [
```

Comment:
I saw that in same rule for store we had an valid tests case which allows to use map operator on select when it was in scope of an effect, it's name is "OkBecauseOfEffect". Does we should respect mapping in scope of component store effects to? If yes maybe it's worth adding test case for that?

Reply:
Good question. 
I didn't add this here because the use-case is different. 
Within an effect that uses the global store there are some cases where you'd want to read a different slice of the state for a particular effect.

I don't think this is a good practice to do with component store effects. Because the state is already scoped.
For the rare cases you can use `get` to read the state, https://ngrx.io/guide/component-store/read#get-method in a component store effect.